app/workers/mining_worker.rb

```diff
@@ -20,100 +13,101 @@ def perform(player_id, asteroid_id, is_mining = false, check_count = 0)
       end
 
       # Add to mission if user has active mission
       mission = player.missions.where(mission_loader: "asteroid.#{asteroid.asteroid_type}_ore", mission_status: 'active', mission_type: 'mining').where("mission_amount > 0").first rescue nil
       if mission
-        mission.update_columns(mission_amount: mission.mission_amount - mining_amount)
-        mission.update_columns(mission_amount: 0) if mission.mission_amount < 0
+        mission.update(mission_amount: mission.mission_amount - mining_amount)
+        mission.update(mission_amount: 0) if mission.mission_amount < 0
       end
 
       # Log
-      ac_server.broadcast("player_#{player_id}", method: 'log', text: I18n.t('log.you_mined_from_asteroid', amount: mining_amount, ore: Item.get_attribute("asteroid.#{asteroid.asteroid_type}_ore", :name).downcase))
+      player.broadcast(:log, text: I18n.t('log.you_mined_from_asteroid', amount: mining_amount, ore: Item.get_attribute("asteroid.#{asteroid.asteroid_type}_ore", :name).downcase))
 
       # Get enemy
       EnemyWorker.perform_async(nil, player.location.id) if rand(10) == 9
 
       # Restart MiningWorker
-      MiningWorker.perform_async(player.id, asteroid_id, true) && (return)
+      MiningWorker.perform_async(player.id, asteroid_id, true)
+      return
 
     end
   end
 
   def can_mine(player, asteroid, mining_amount)
     player = player.reload
-    asteroid = asteroid.reload rescue nil
-
-    # Set Variables
-    player_id = player.id
-    asteroid_id = asteroid.id rescue nil
+    asteroid = asteroid&.reload
 
     # Remove asteroid as target if depleted
     if asteroid.resources <= 0
       # Tell other users who miner this rock is depleted
-      User.where(mining_target_id: asteroid_id).where("online > 0").each do |u|
-        ActionCable.server.broadcast("player_#{u.id}", method: 'asteroid_depleted')
+      User.where(mining_target: asteroid).is_online.each do |u|
+        u.broadcast(:asteroid_depleted)
+        # Q: shouldn't we clear their mining targets as well?
+        #    and isn't the player included in this list?
       end
 
       asteroid.destroy
-      ActionCable.server.broadcast("player_#{player_id}", method: 'asteroid_depleted')
-
-      player.update_columns(mining_target_id: nil)
+      player.broadcast(:asteroid_depleted)
+      player.update(mining_target_id: nil)
       return false
     end
 
     # Stop mining if user has other or no mining target
-    if player.mining_target_id != asteroid_id || player.target_id != nil
+    if player.mining_target != asteroid || player.target_id != nil
       return false
     end
 
     # Stop mining if player's ship is full
     if player.active_spaceship.get_free_weight <= 0
-      ActionCable.server.broadcast("player_#{player_id}", method: 'asteroid_depleted')
-      player.update_columns(mining_target_id: nil)
+      # Q: if the ship is full, is the asteroid really depleted?
```

Comment:
The asteroid is not depleted but the `:asteroid_depleted` broadcast currently only removes the players mining target in the client, which is why I recycle it here. Probably would be better to create a broadcast with a more matching name.